### PR TITLE
Endpoint for receipt delivery

### PR DESCRIPTION
**Pull request by enahum**
Add endpoint for receipt delivery (aka acknowledge), including metrics and the sender name for the push notification

https://mattermost.atlassian.net/browse/MM-14858
https://mattermost.atlassian.net/browse/MM-14884

### CODE REVIEW

File: server/apple_notification_server.go [hunk 1]
```diff
@@ -94,6 +98,10 @@ func (me *AppleNotificationServer) SendNotification(msg *PushNotification) PushR
 		data.Custom("sender_id", msg.SenderId)
 	}
 
+	if len(msg.SenderName) > 0 {
```
Comment (jespino) [inline server/apple_notification_server.go]: Looks not related stuff, but I don't see any problem to include this new data as part of the PR if that is on purpose :)

**Comment by enahum** [comment]
@jespino is part of another ticket done in the same PR, you’ll see the reference once you review the server PR ;)

File: server/server.go [hunk 2]
```diff
@@ -153,6 +156,47 @@ func handleSendNotification(w http.ResponseWriter, r *http.Request) {
...
+
+	if ack == nil {
+		rMsg := LogError("Failed to read ack body")
+		w.Write([]byte(rMsg.ToJson()))
+		incrementBadRequest()
+		return
+	}
+
+	if len(ack.Id) == 0 {
+		// TODO: Do we need the ID? we are not really tracking it here
```
Comment (hmhealey) [inline server/server.go]: Did you mean to leave this TODO in?

Comment (enahum) [reply server/server.go]: removed

**Pull request merged**